pkg/operator/ceph/cluster/cluster_external.go

```diff
@@ -112,6 +113,20 @@ func (c *ClusterController) configureExternalCephCluster(cluster *cluster) error
 		return errors.Wrap(err, "failed to create csi config map")
 	}
 
+	// update the msgr2 flag
+	for _, m := range cluster.ClusterInfo.Monitors {
+		// m.Endpoint=10.1.115.104:3300
+		monPort := util.GetPortFromEndpoint(m.Endpoint)
+		if monPort == client.Msgr2port {
+			if cluster.Spec.Network.Connections == nil {
+				cluster.Spec.Network.Connections = &cephv1.ConnectionsSpec{}
+			}
+			cluster.Spec.Network.Connections.RequireMsgr2 = true
+			logger.Debugf("one of the mons has v2 port, so RequireMsgr2 flag is required to set")
```

Comment:
We're not actually changing the RequireMsgr2 flag on the cluster CR, it's just an implementation detail, so perhaps this is more clear:
```suggestion
			logger.Debugf("a v2 port was found for a mon endpoint, so msgr2 is required")
```

Reply:
Updated, thanks